BE/app/Http/Repositories/UniversitySearch/UniversitySearchRepository.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Http\Repositories\UniversitySearch;
+
+use App\Http\Repositories\Traits\Helpers;
+use Illuminate\Support\Facades\Cache;
+
+/**
+ * Class UniversitySearchRepository
+ * @package App\Http\Repositories\UniversitySearch
+ */
+class UniversitySearchRepository implements UniversitySearchRepositoryInterface
+{
+    use Helpers;
+
+    const RESULT_LENGTH = 5;
+    private ?string $searchParam = null;
+
+    /**
+     * @param string|null $searchParam
+     * @return array
+     * @throws \Exception
+     */
+    public function search(?string $searchParam = null): array
+    {
+        try {
+            return $this->setSearchParam($searchParam)
+                        ->getResult();
+        } catch (\Exception $exception) {
+            throw $exception;
+        }
+    }
+
+    /**
+     * @return array
+     */
+    protected function getResult(): array
+    {
+        if (empty(Cache::get('universities'))) {
```

Comment:
This is incorrect. The cache-aside pattern means the cache should be populated on request.

Reply:
It Cache of Laravel and this part of code check if we have cached data with universities key.There wasn't detailed information in the task about how I should cache data from external api, and I decided to use command for cache external api data, and already fetch data from cache on request. I did not see anything incorrect in this part of code.

Reply:
You're right - the previous version of the task requested a particular cache strategy. In this one:

```
. Data should be cached on the server and client sides
. You must choose the caching strategy and method
. Please explain your choice in the README
. Caching TTL should be configurable
```

I suppose I wouldn't have expected this strategy though, because if the source data changes the command would need to be re-run which is probably not practical in a real app.

Reply:
I agree, in a real project I myself would not have chosen this approach, since data from the API can always be updated, but in this case, since this was a test task and I could choose the approach myself, I used the console command